test/syntax.jl

```diff
@@ -2345,3 +2345,9 @@ end
 
 # issue #37656
 @test :(if true 'a' else 1 end) == Expr(:if, true, quote 'a' end, quote 1 end)
+
+# issue #37664
+@test_throws ParseError("extra token \"b\" after end of expression") Meta.parse("a b")
```

Comment:
Could you add a test in a space-sensitive context as well, like a macro call or an hcat expression?